### PR TITLE
[node-manager] add requirements for nodes

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
Signed-off-by: Denis Romanenko <denis.romanenko@flant.com>

## Description

This is part of https://github.com/deckhouse/deckhouse/pull/3388, which be implemented before main part.
## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: chore
summary: Add requirements for nodes
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+			value, exists := requirements.GetValue(minVersionValuesKey)
+			Expect(exists).To(BeTrue())
+			Expect(value).To(BeEquivalentTo("20.4.3"))
+		})
+	})
+
+	Context("One node with Centos OS and requirements set", func() {
+		BeforeEach(func() {
+			requirements.SaveValue(minVersionValuesKey, "1.2.3")
+			f.BindingContexts.Set(f.KubeStateSet(node3))
```
Comment (shvgn) [inline modules/040-node-manager/hooks/minimal_node_os_version_test.go:120]: My wish would be to rename node constants, so here I would not need to scroll upwards to see what the node is really about.

```
node1 → nodeUbuntu2004
node2 → nodeUbuntu1804
node3 → nodeCentos7
```

**Review by shvgn — DISMISSED** [review]
LGTM

File: modules/040-node-manager/hooks/minimal_node_os_version_test.go (hunk 1 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/hooks/minimal_node_os_version_test.go]: done

**Review by diafour — APPROVED** [review]

**Review by shvgn — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Pull request merged**